src/env.rs

```diff
@@ -141,6 +144,11 @@ impl Environment {
                 shm_formats2.lock().unwrap().push(format);
             });
 
+        let data_device_manager = manager
+            .instantiate_auto::<wl_data_device_manager::WlDataDeviceManager>()
+            .expect("Server didn't advertize `wl_data_device_manager`?!")
```

Comment:
So this will make winit apps panic on smithay right now, doesn't it? :D 
Why don't you treat at least some of the objects of env optional?

Reply:
Well, yes, if I release a new version of SCTK right away, it will. ^^

The main reason I didn't make them optional is that they are part of the core protocol: any compositor is supposed to support them. As such, it allows me to remove the burden of handling an `Option<_>` from the API.

Though, it may be possible to add some sneaky logic to lazily instantiate these globals upon use, I need to think about it. But this would be the scope for an other PR I think.